### ISSUE TITLE
[Backport ] [Enhancement] add probe customization to the Rancher chart

### ISSUE THREAD

**Issue by rancherbot** [rancher/rancher#39942]
This is a backport issue for https://github.com/rancher/rancher/issues/39841, automatically created via rancherbot by @cbron

Original issue description:

Breakout from: https://github.com/rancher/rancher/issues/38177
Adds values to the chart for rancher server to override the livenessProbe and readinessProbe's initialDelaySeconds and periodSeconds

**Comment by sowmyav27** [comment]
Should be available to validate in `v2.6.11-rc1`

**Comment by brudnak** [comment]
# ✅ PASSED

## Reproduction Environment

#### Not required.

<hr>

## Validation Environment


| Component                        | Version / Type           |
| -------------------------------- | ------------------------ |
| Rancher version                  | v2.6.11-rc2              |
| Installation option              | Helm (high availability) |
| RKE binary version used          | v1.3.15                  |
| If Helm Chart k8s cluster        | v1.24.4                  |
| Cert Details                     | Let's Encrypt / `nginx`  |
| Docker version                   | 20.10.7, build f0df350   |
| Helm version                     | v2.16.8-rancher2         |
| Downstream cluster type          | not applicable           |
| Downstream K8s version           | not applicable           |
| Authentication providers enabled | local                    |
| Logged in user role              | Administrator            |
| Browser type                     | Google Chrome            |
| Browser version                  | 109.0.5414.119 (x86_64)  |

<details>
    <summary>🚨 Additional Validation Setup Details: Click to Expand</summary>

```shell
#!/bin/sh

helm repo update

kubectl create namespace cattle-system

kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.7.1/cert-manager.crds.yaml

helm install cert-manager jetstack/cert-manager \
  --namespace cert-manager \
  --create-namespace \
  --version v1.7.1

helm install rancher rancher-latest/rancher \
  --namespace cattle-system \
  --set hostname=$EMAIL_VAR \
  --set bootstrapPassword=$PW_VAR \
  --set ingress.tls.source=letsEncrypt \
  --set letsEncrypt.email=$EMAIL_VAR \
  --set letsEncrypt.ingress.class=nginx \
  --set rancherImageTag=v2.6.11-rc2 \
  --version 2.6.11-rc2 \
  --set livenessProbe.initialDelaySeconds=42 \
  --set livenessProbe.periodSeconds=42 \
  --set readinessProbe.initialDelaySeconds=42 \
  --set readinessProbe.periodSeconds=42 \
  --set startupProbe.periodSeconds=42

```

</details>

## Validation steps

1. Setup Rancher server with the local cluster as RKE1
2. When doing helm install provide the additional set values

```shell
  --set livenessProbe.initialDelaySeconds=42 \
  --set livenessProbe.periodSeconds=42 \
  --set readinessProbe.initialDelaySeconds=42 \
  --set readinessProbe.periodSeconds=42 \
  --set startupProbe.periodSeconds=42
```

3. Once Rancher is installed >>> navigate to local cluster Kubectl Shell
4. Run the following command

```shell
kubectl get pods -n cattle-system
kubectl get pod <rancher-pod-name-here> -n cattle-system -o yaml
```

5. Verify the probe settings have been updated
6. Everything was updated as expected, see screenshot below

## Additional Info


![2023-02-06_14-36-54](https://user-images.githubusercontent.com/40877149/217125462-94dc8c55-6e51-4482-9c6f-edd5c4ac3364.png)


## RESULTS

#### ✅ Expected
For the supplied values to be set

#### ✅ Actual
The supplied values were successfully set
